plugins/basic/python/tk_maya_basic/plugin_logic.py

```diff
@@ -334,3 +345,40 @@ def _jump_to_signup():
     QtGui.QDesktopServices.openUrl(
         QtCore.QUrl("https://www.shotgunsoftware.com/signup")
     )
+
+
+class Callback(object):
+    def __init__(self, callback):
+        self.callback = callback
+
+    def __call__(self, *args):
+        """
+        Execute the callback deferred to avoid potential problems with the command resulting in the menu
+        being deleted, e.g. if the context changes resulting in an engine restart! - this was causing a
+        segmentation fault crash on Linux.
+
+        :param state: The state of the menu item.
+        :return: None
+        """
+        # note that we use a single shot timer instead of cmds.evalDeferred as we were experiencing
+        # odd behaviour when the deferred command presented a modal dialog that then performed a file
+        # operation that resulted in a QMessageBox being shown - the deferred command would then run
+        # a second time, presumably from the event loop of the modal dialog from the first command!
+        #
+        # As the primary purpose of this method is to detach the executing code from the menu invocation,
+        # using a singleShot timer achieves this without the odd behaviour exhibited by evalDeferred.
+
+        # This logic is borrowed from the menu_generation.py Callback class.
+
+        QtCore.QTimer.singleShot(0, self._execute_within_exception_trap)
+
+    def _execute_within_exception_trap(self):
+        """
+        Execute the callback and log any exception that gets raised which may otherwise have been
+        swallowed by the deferred execution of the callback.
+        """
+        try:
+            self.callback()
+        except Exception:
+            current_engine = sgtk.platform.current_engine()
+            current_engine.logger.exception("An exception was raised from Toolkit")
```

Comment:
How about this?

```suggestion
            current_engine.logger.exception("An exception was raised from a menu item:")
```
Looking at the stack trace it will be easy to figure out it's coming from Toolkit, and by the message we'll have a feeling in what part of the engine it broke.

Reply:
Yeah seems sensible! Thanks